pinot-spi/src/main/java/org/apache/pinot/spi/accounting/ThreadAccountant.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.accounting;
+
+import javax.annotation.Nullable;
+
+
+public interface ThreadAccountant {
+
+  /**
+   * clear thread accounting info
+   */
+  void clear();
+
+  /**
+   * check if the corresponding runner thread of current thread is interrupted
```

Comment:
Suggest calling it anchor / parent thread since runner-worker terminology is applicable on server and this framework is ideally going to be reusable for broker solution as well.

Reply:
Good point, updated

Reply:
Changed all naming of root thread to anchor thread